compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3073,6 +3073,7 @@ BRecordType createRecordTypeForRestField(Location pos, SymbolEnv env, BRecordTyp
                                        BType restConstraint) {
         BRecordTypeSymbol recordSymbol = createAnonRecordSymbol(env, pos);
         BRecordType recordVarType = new BRecordType(recordSymbol);
+        recordSymbol.type = recordVarType;
```

Comment:
According to this fix, the NPE occurs when attempting to access the type from its type symbol. My concern is why are we attempting to access the type through its type symbol? Is it not possible to directly utilize the type without going through its symbol?

Reply:
Here we are accessing the owner through the `RecordTypeSymbol`